.travis-build.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+# This script compiles Guava, performing type-checking.
+
+# If optional argument $1 is "-P checkerframework-local",
+# use a locally-built clone of the Checker Framework repository.
+export MVN_FLAGS=$1
+
+# Fail the whole script if any command fails.
+set -e
+
+## Diagnostic output
+# Output lines of this script as they are read.
+set -o verbose
+# Output expanded lines of this script as they are executed.
+set -o xtrace
+
+export SHELLOPTS
+
+SLUGOWNER=${TRAVIS_PULL_REQUEST_SLUG%/*}
+if [[ "$SLUGOWNER" == "" ]]; then
+  SLUGOWNER=${TRAVIS_REPO_SLUG%/*}
+fi
+if [[ "$SLUGOWNER" == "" ]]; then
+  SLUGOWNER=typetools
+fi
+echo SLUGOWNER=$SLUGOWNER
+
+echo TRAVIS_PULL_REQUEST_BRANCH = $TRAVIS_PULL_REQUEST_BRANCH
+echo TRAVIS_BRANCH = $TRAVIS_BRANCH
+
+## Build Checker Framework
```

Comment:
There is no reason to build the Checker Framework in this Travis build if the `-P checkerframework-local` flag is not going to be passed to maven, correct?

If so, why not guard all of this with a check against that flag?

Reply:
You are correct that building the Checker Framework is unnecessary if it won't be used, and that this wastes time in some/many jobs.

The flag can be arbitrary, so determining from the flag whether the Checker Framework is not a simple check and I didn't want to hard-code a brittle or error-prone check.